packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -88,6 +88,14 @@ interface DatabaseClusterBaseProps {
    */
   readonly deletionProtection?: boolean;
 
+  /**
+   * Whether to enable mapping of AWS Identity and Access Management (IAM) accounts
+   * to database accounts.
+   *
+   * @default false
+   */
+  readonly iamAuthentication?: boolean;
```

Comment:
I'm wondering: why add this property to `DatabaseClusterBaseProps`? This means this is also available for the `DatabaseClusterFromSnapshot` resource. Is this correct? Can `iamAuthentication` be enabled for Clusters created from snapshots?

Maybe this property should be added to `DatabaseClusterProps` instead?

Reply:
Wrote a test case for `DatabaseClusterFromSnapshot`. Worked fine with the existing changes

Reply:
I understand a unit test work.

Can you try deploying a `DatabaseClusterFromSnapshot` with this property set, and let me know if it deploys without problems?